bin/autoaggregate

```diff
@@ -53,27 +93,13 @@ def aggregate(config):
             cwd=SRC_DIR,
             stderr=sys.stderr,
             stdout=sys.stdout,
+            preexec_fn=pre_exec,
         )
     finally:
-        # Restore umask, if changed
-        if old_umask is not None:
-            os.umask(old_umask)
         # Chown recursively, if UID or GID are specified
+        # just to make sure everything is owned by odoo:odoo (should run empty)
```

Comment:
This comment is probably fake (see below).

Reply:
What do you mean by fake comment (so i can improve it)?

Reply:
Yes, sorry for the bare explanation.

I mean that not everything should be owned by `odoo:odoo`, but by `$UID:$GID`, in case they're specified. That's what the previous comment line says, and probably you were trying this chown because of https://github.com/Tecnativa/doodba/pull/413#discussion_r612978447.

So probably this comment is wrong.

Reply:
I was trying this chown because that's how it worked before (as long as i didn't missinterprete the code before):
* checkout everything as root:root in autoaggregate (by using root for gitaggregate), writing the data once
* chowning to odoo:odoo (`$UID:$GID`) with python in autoaggregate, writing the data twice
* later chowning to root:odoo with 800-permissions, writing the data a third time
And ideally with the fix/workarround the fix_permissions would not be needed anymore in aggregate (when we set the right permissions from the beginning). But unfortunately it still does change the .git (pack if i remeber correctly) readonly files from `r--r-----` to `rw-r-----`. Getting rid of this would make the code for fix_permissions even more complicated and unreadable and the performance impact was less than a second on our slow systems so it seemed not worth it.

So i would (if you agree that this is the right way)
* set the UID to 0 here for gitaggregate (to make everything owned by user root, and using git config from root)
* keep the GID code (to set the correct group right away, so everything should be assigned to group odoo)
* adapt  the comments (explaining that ideally fix_permissions should do nothing at this point)
* also i would try to make the doodba DEBUG log level switch own the -c output for chmod and -v output for chown

Reply:
Ok had some time to think the plan i mentioned above does not work:
* autoaggregate is also invoked with setup-devel.yaml to checkout locally, and permissions there need to match UID:GID of the invoking user, otherwhise the user is not the owner of his own files

How would you feel about another user (having a checkout user that can be the same UID as the developer but a different UID from the runtime odoo user (sharing the odoo group together))?